File: server/basemodels/medical_institution.py
from typing import List

from pydantic import BaseModel


# Medical Institution
class MedicalInstitutionGetResponse(BaseModel):
    id: str
    name: str
    mfl_code: str | None = None
    dhis_code: str | None = None
    county: str | None = None
    sub_county: str | None = None


class MedicalInstitutionPostRequest(BaseModel):
    name: str
    mfl_code: str | None = None
    dhis_code: str | None = None
    county: str | None = None
    sub_county: str | None = None


# Medical Institution Telephone
class MedicalInstitutionTelephoneGetResponse(BaseModel):
    medical_institution_id: str
    telephone: str


class MedicalInstitutionTelephonePostRequest(BaseModel):
    medical_institution_id: str
    telephone: str


class MultipleMedicalInstitutionTelephonePostRequest(BaseModel):
    telephones: List[MedicalInstitutionTelephonePostRequest]

File: client/app/types/medical_institution.ts
export interface MedicalInstitutionPostRequestInterface {
	name: string;
	mfl_code?: string;
	dhis_code?: string;
	county?: string;
	sub_county?: string;
}

export interface MedicalInstitutionPostResponseInterface {
    id: string;
	name: string;
	mfl_code?: string;
	dhis_code?: string;
	county?: string;
	sub_county?: string;
}

export interface MedicalInstitutionGetResponseInterface {
    id: string;
	name: string;
	mfl_code?: string;
	dhis_code?: string;
	county?: string;
	sub_county?: string;
}